src/wp-includes/global-styles-and-settings.php

```diff
@@ -342,6 +332,47 @@ static function ( $item ) {
 	}
 }
 
+/**
+ * Gets the block name from a given theme.json path.
+ *
+ * @since 6.3.0
+ * @access private
+ *
+ * @param array $path An array of keys describing the path to a property in theme.json.
+ * @return string Identified block name, or empty string if none found.
+ */
+function wp_get_block_name_from_theme_json_path( $path ) {
+	// Block name is expected to be the third item after 'styles' and 'blocks'.
+	if (
+		count( $path ) >= 3
+		&& 'styles' === $path[0]
+		&& 'blocks' === $path[1]
+		&& str_contains( $path[2], '/' )
+	) {
+		return $path[2];
+	}
+
+	/*
+	 * As fallback and for backward compatibility, allow any core block to be
+	 * at any position.
+	 */
+	$result = array_values(
+		array_filter(
+			$path,
+			static function ( $item ) {
+				if ( str_contains( $item, 'core/' ) ) {
```

Comment:
Just to check: All of the tests with an element containing `core/` starts with it. Is there a case where a core block would contain `core/` but not start with it?

I'm thinking about clashes with extender block names, like `blockcore/blockymcblockface`.

Reply:
Just saw your earlier comment regarding this being pre-existing code. That's fair enough - though it does look prone to possible clashes with extender blocks :thinking_face:

Reply:
Could potentially fix it in another ticket. Maybe too late for 6.3 though.